x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/SecondaryAuthenticator.java

```diff
@@ -76,7 +76,11 @@ public void authenticateAndAttachToContext(RestRequest request, ActionListener<S
         // Use cases for secondary authentication are far more likely to want to fall back to the primary authentication if no secondary
         // auth is provided, so in that case we do no want to set anything in the context
         authenticate(
-            authListener -> authenticationService.authenticate(request, false, authListener),
+            authListener -> authenticationService.authenticate(
+                request,
+                false,
+                ActionListener.runBefore(authListener, () -> authenticationService.auditSuccess(request))
```

Comment:
Secondary authentication must also be audited.

Reply:
`runBefore` runs onFailure path as well. I think that's not what we want here?

Reply:
Good catch! Thanks! Pushed [e14ddfe](https://github.com/elastic/elasticsearch/pull/94120/commits/e14ddfea7f02fdf07dd064c42221a75639c005e8)